### ISSUE TITLE
Crash when using stackable conveyors 0.16

### ISSUE THREAD

**Issue by ek0100** [Algalish/SmartIssueTracker#122]
Everything works great when creating belts with the stackable conveyors, but I run into this crash whenever I first add items onto the belts.  It seems to work okay after reloading

Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:d:\ws\sb-201123102323-c9f\ue4\engine\source\runtime\core\public\Containers/Array.h] [Line: 611] Array index out of bounds: 25 from an array of size 25

**Comment by Algalish** [comment]
@ek0100  I had this problem but did a lot of tests and was not able to repeat it.
ok, so now I know that it is still present. will try to fix it asap

**Comment by ek0100** [comment]
It's oddly inconsistent, and difficult to reproduce...I'll reload and perform the same steps that led to the crash, and it works normally...but if I splice onto a different belt, or the same belt at a different location, it may cause the crash again.  I'll update if I notice a pattern or anything that may help troubleshoot this.  Thanks for the help!

**Comment by velillen** [comment]
Ive been having basically the same problem. This seems to cause it every time for me...

1) Stackable Conveyors, "4" high, decently long (over 10 "sections")
2) Manual belting to hook up to resource node
3) Soon as hooked up, once the resource hits the belting that was made with the mod (step 1), it crashes.
Of note..if make the stack able Conveyors with the mod (ie step 1) and then saving the game and restarting the game, then the problem doesnt seem to occur.

Error code
Assertion failed: (Index >= 0) & (Index < ArrayNum) [File:d:\ws\sb-201123102323-c9f\ue4\engine\source\runtime\core\public\Containers/Array.h] [Line: 611] Array index out of bounds: 16 from an array of size 16



Hope that helps.

**Comment by ek0100** [comment]
Yeah, my experience is identical to @velillen.  It seems that saving then reloading the save will avoid the crash without needing to restart the game completely.

**Comment by Yoeghoert** [comment]
Same here,
Have been testing some stuff to avoid it. Without any succes.
The stackable conveyor poles have a placing 'direction' but that doesn't have anything to do with it.
It seems to be triggering on the moment that the added material starts stacking on the belt due to being full.
Once your SMART belt length is fully filled the game crashes.

Best to prep all belts. Save game, reload and than link all product filled belts.

**Comment by Algalish** [comment]
@ek0100 @Yoeghoert @velillen 
After countless attempts it looks like there is nothing I can do about it. Because it is not directly crashing for the mod but for the game itself.
I reached out to CSS and they agreed to take a look at that.
There are 2 types of crashes. They said that one of them is already fixed in EXP (so will be fixed when I update the mod for update 4 of the game)
But the other one - they will look into it.

For me it is crashing 100% so I cannot use this feature with stackable poles. This is very frustrating but we will get it fixed
